src/index.js

```diff
@@ -4,23 +4,77 @@
  * Module dependencies.
  */
 
+const { serializeError } = require('serialize-error');
+const cloneDeep = require('lodash/cloneDeep');
 const get = require('lodash.get');
 const stringify = require('json-stringify-safe');
 const traverse = require('traverse');
+const update = require('lodash/update');
 
 /**
  * Constants.
  */
 
 const DEFAULT_REPLACEMENT = '--REDACTED--';
+const defaultSerializers = {
+  'error-serializer': serializeError
+};
+
+/**
+ * Apply serializers.
+ */
+
+function applySerializers(values, serializers) {
+  const data = cloneDeep(values);
+
+  for (const { path, serializer } of serializers) {
+    let resolvedSerializer;
+
+    if (typeof serializer === 'function') {
+      resolvedSerializer = serializer;
+    } else if (typeof serializer === 'string' && serializer in defaultSerializers) {
+      resolvedSerializer = defaultSerializers[serializer];
+    } else {
+      throw new TypeError(
+        `Invalid serializer for \`${path}\` path: must be a function or a key (string) in the list of default serializers (${Object.keys(
+          defaultSerializers
+        )})`
+      );
+    }
```

Comment:
Serializers should be checked and resolved at invocation, not every time we need to anonymize some object.